### ISSUE TITLE
Multiple distinct graphs

### ISSUE THREAD

**Issue by hjoliver** [cylc/cylc-flow#4903]
This comes up fairly regularly, but we don't seem to have an issue for it.

Many (most?) workflows have "prep" tasks in the first cycle that need to finish before any future-cycle tasks begin. And similarly, there's often a need for special clean up tasks in the final cycle point, after everything else has finished.

We can do this easily enough (use prep tasks as an example) with perpetual dependence on the initial cycle point:
```ini
R1 = "prep"
P1 = "prep[^] => foo => bar"
```
or with artificial dependencies to force the first tasks in each cycle to start in order:
```ini
R1 = "prep => foo"
P1 = "foo[-P1]:submit => foo => bar"
```
However, neither of these solutions is very elegant. The first is correct in principle, but it seems a bit silly retain the `^` dependency forever, and it has performance implications for long-running workflows due to increasingly long datetime sequence computations. The second is artificial and unnecessarily prevents cycles from running out of order.

What users really want here is just a separate "start-up graph" that runs to completion before the main cycling graph starts.

### Proposal A: allow multiple sequential `graph`s under the scheduling section

```ini
[scheduling]
   cycling mode = integer
   initial cycle point = 1
   [[graph_1]]
        R1 = "prep"
   [[graph_2]]
        P1 = "foo => bar" 
   [[graph_3]]
        R1 = "cleanup"
```

### Proposal B: allow multiple sequential `[scheduling]` sections

Multiple `graph`s would probably suffice for most use cases, but for more power and flexibility we should have separate control of cycling,  queues, etc., for each of the distinct workflow graphs.

```ini
[scheduling_1]
   [[graph]]
        R1 = "prep"

[scheduling_2]
   cycling mode = integer
   initial cycle point = 1
   [[graph]]
        P1 = "foo => bar" 

[scheduling_3]
   [[graph]]
        R1 = "cleanup"
```

**Comment by hjoliver** [comment]
I think this would be quite easy to implement. The scheduler only needs to manage one graph at a time. If there is nothing more to do (and we're not stalled) shut down if there are no more graphs, or start running the next graph if there is one.

**Comment by hjoliver** [comment]
(Context: in my experience, this is a common problem for new users: how do I stop cycle 2 from starting before cycle 1 is finished doing its prep work?)

**Comment by oliver-sanders** [comment]
More generally multiple distinct graphs can be considered subgraphs, this [has previously been discussed](https://github.com/cylc/cylc-flow/wiki/Possible-Cylc-Futures#combining-graphs) in the light of #1962.

**Comment by hjoliver** [comment]
Yes I'm aware this will be solved on the Cylc 9 timescale, but that's probably several years off. I'm suggesting here that the "initial graph" and "final graph" problem can be solved quite easily under the current framework.

**Comment by MetRonnie** [comment]
Imho this is more of a https://github.com/cylc/cylc-flow/labels/speculative issue than https://github.com/cylc/cylc-flow/labels/question

**Comment by hjoliver** [comment]
Well, the question was just should we do this? esp. as after a little though I think it'd be quite easy to do. It's not particularly speculative, as the need for start-up and shutdown-graphs has come up multiple times over the years.

The *multi-dimensional cycling, but the cycles are "a prison"* thing is really not an issue for my intended use case, because that is literally what's wanted for start-up and shutdown graphs. However, for more general multi-dimensional cycling it certainly is an issue, and my proposal could be abused in that way (albeit to a limited extent if we didn't allow an arbitrary number of separate graphs).

However, I'm quite happy to go with @oliver-sanders variable runahead as an interim solution, as that will be super easy to implement and would handle the primary use case (start-up graph) very nicely.

So I'll close this, and open a new issue for that...
